pkg/deploy/deploy.go

```diff
@@ -206,127 +205,28 @@ func GetResources(resMap resmap.ResMap) ([]*unstructured.Unstructured, error) {
-		selector := labels.ODH.Component(componentName)
-		// only removed the resource with our label applied, not the same name resource maually created by user
-		if existingOwnerReferences == nil && resourceLabels[selector] == "true" {
-			return cli.Delete(ctx, found)
-		}
-
-		found.SetOwnerReferences([]metav1.OwnerReference{})
-		data, err := json.Marshal(found)
-		if err != nil {
-			return err
-		}
-
-		err = cli.Patch(ctx, found, client.RawPatch(types.ApplyPatchType, data), client.ForceOwnership, client.FieldOwner(owner.GetName()))
-		if err != nil {
-			return err
-		}
-
-		return cli.Delete(ctx, found)
-	}
-
-	// Create the resource if it doesn't exist and component is enabled
-	if apierrs.IsNotFound(err) {
-		// Set the owner reference for garbage collection
-		// Skip set on CRD, e.g. we should not delete notebook CRD if we delete DSC instance
-		// Skip on OdhDashboardConfig CR, because we want user to be able to update it
-		if found.GetKind() != "CustomResourceDefinition" && found.GetKind() != "OdhDashboardConfig" {
-			if err = ctrl.SetControllerReference(owner, metav1.Object(obj), cli.Scheme()); err != nil {
-				return err
-			}
-		}
-
-		return cli.Create(ctx, obj)
-	}
-
-	// Exception: ODHDashboardConfig should not be updated even with upgrades
-	// TODO: Move this out when we have dashboard-controller
-	if found.GetKind() == "OdhDashboardConfig" {
-		// Do nothing, return
-		return nil
-	}
-
-	// Kserve specific workflow:
-	// TODO: Remove this when we have generalize custom config requirements across all components
-	if componentName == "kserve" {
-		// do not reconcile kserve resource with annotation "opendatahub.io/managed: false"
-		if found.GetAnnotations()[annotation.ManagedByODHOperator] == "false" {
-			return nil
-		}
-		// do not patch resources field in Kserve deployment i.e allows users to update resources field
-		if err := removeResourcesFromDeployment(obj); err != nil {
-			return err
-		}
-	}
-
-	// JIRA 6889.  odh-model-controller will be covered by either kserve or model-mesh case
-	if componentName == "model-mesh" {
-		if err := removeResourcesFromDeployment(obj); err != nil {
-			return err
-		}
+	// Return if error getting resource in cluster
+	found, err := getResource(ctx, cli, obj)
+	if client.IgnoreNotFound(err) != nil {
+		return err
 	}
 
-	// Preserve app.opendatahub.io/<component> labels of previous versions of existing objects
-	foundLabels := make(map[string]string)
-	for k, v := range found.GetLabels() {
-		if strings.Contains(k, labels.ODHAppPrefix) {
-			foundLabels[k] = v
-		}
+	if !enabled {
+		return handleDisabledComponent(ctx, cli, found, componentName, owner)
 	}
-	newLabels := obj.GetLabels()
-	maps.Copy(foundLabels, newLabels)
-	obj.SetLabels(foundLabels)
 
-	// Perform server-side apply
-	data, err := json.Marshal(obj)
-	if err != nil {
-		return err
+	// Create resource if it doesn't exist
+	if apierrs.IsNotFound(err) || found == nil {
```

Comment:
This found check looks redundant. If the object lookup returns NotFound, it's created.  Creation uses only manifest's object. (If it's not NotFound, line 216 returns. So, updateResource happens only when no error).